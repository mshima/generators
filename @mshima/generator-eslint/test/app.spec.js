import path, { dirname } from 'path';
import expect from 'expect';
import helpers from 'yeoman-test';

import { fileURLToPath } from 'url';

const __dirname = dirname(fileURLToPath(import.meta.url));

describe('@mshima/eslint:app', () => {
  let runContext;
  let runResult;

  beforeEach(async () => {
    runContext = helpers.create('@mshima/eslint:app').withLookups([{ npmPaths: path.join(__dirname, '..', '..') }]);
  });

  afterEach(() => {
    runContext.cleanTestDirectory();
  });

  describe('with sample values', () => {
    beforeEach(async () => {
      runResult = await runContext.run();
    });

    it('should match snapshot', () => {
      expect(runResult.getSnapshot()).toMatchSnapshot();
    });
  });
});
