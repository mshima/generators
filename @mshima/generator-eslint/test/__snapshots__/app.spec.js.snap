// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@mshima/eslint:app with sample values should match snapshot 1`] = `
Object {
  ".eslintrc.json": Object {
    "contents": "{
  \\"env\\": {
    \\"node\\": true,
    \\"es6\\": true
  },
  \\"extends\\": [\\"eslint:recommended\\", \\"plugin:import/recommended\\"],
  \\"parserOptions\\": {
    \\"ecmaVersion\\": 2020
  }
}
",
    "stateCleared": "modified",
  },
  "package.json": Object {
    "contents": "{
  \\"scripts\\": {
    \\"pretest\\": \\"concurrently -m 1 npm:pretest:*\\",
    \\"pretest:eslint\\": \\"npm run eslint:check\\",
    \\"fix\\": \\"concurrently -m 1 npm:fix:*\\",
    \\"fix:eslint\\": \\"npm run eslint:fix\\",
    \\"eslint:fix\\": \\"eslint --fix .\\",
    \\"eslint:check\\": \\"eslint .\\"
  },
  \\"devDependencies\\": {
    \\"concurrently\\": \\"^4.1.0\\",
    \\"eslint\\": \\"7.26.0\\",
    \\"eslint-plugin-import\\": \\"2.23.2\\"
  }
}
",
    "stateCleared": "modified",
  },
}
`;
